synapseclient/models/permission.py

```diff
@@ -0,0 +1,175 @@
+from dataclasses import dataclass
+from opentelemetry import trace
+
+tracer = trace.get_tracer("synapseclient")
+
+
+@dataclass
+class Permissions:
+    """The permission a User has for a given Entity.
+
+
+    Attributes:
+        can_view : Can the user view this entity?
+        can_edit : Can the user edit this entity?
+        can_move : (Read Only) Can the user move this entity by changing its parentId?
+        can_add_child : Can the user add a child entity to this entity?
+        can_certified_user_edit : (Read Only) Can the user edit this entity once they become a Certified User?
+        can_certified_user_add_child : (Read Only) Can the user add a child entity to this entity once they become a Certified User?
+        is_certified_user : (Read Only) True, if the user has passed the user certification quiz.
+        can_change_permissions : Can the user change the permissions of this entity?
+        can_change_settings : Can the user change the settings of this entity?
+        can_delete : Can the user delete this entity?
+        can_download : Are there any access requirements precluding the user from downloading this entity?
+        can_upload : (Read Only) Are there any access requirements precluding the user from uploading into this entity (folder or project)?
+        can_enable_inheritance : (Read Only) Can the user delete the entity's access control list (so it inherits settings from an ancestor)?
+        owner_principal_id : (Read Only) The principal ID of the entity's owner (i.e. the entity's 'createdBy')
+        can_public_read : (Read Only) Is this entity considered public?
+        can_moderate : Can the user moderate the forum associated with this entity? Note that only project entity has forum.
+        is_certification_required : (Read Only) Is the certification requirement enabled for the project of the entity?
+        is_entity_open_data : (Read Only) Returns true if the Entity's DateType equals 'OPEN_DATA', indicating that the data is safe to be released to the public.
+    """
+
+    can_view: bool = None
```

Comment:
When you get a chance take a peek at the SonarCloud results. One of the things it had flagged is this line:
https://sonarcloud.io/project/issues?id=Sage-Bionetworks_synapsePythonClient&pullRequest=1037&resolved=false&inNewCodePeriod=true

`Replace the type hint "bool" with "Optional[bool]" or don't assign "None" to "can_view"`

The import for this is:
`from typing import Optional`.

The code would end up like:
`can_view: Optional[bool] = None`